---
title: "Features "
author: "Byungju Kim"
date: "2018년 11월 9일"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## Featues

Since our genotype data would be categorical, thinking how to handle it is useful.

### from biological knowledge

* Dominant modeling : "AA + AB" vs "BB"

* Recessive Model   : "AA + AB vs AB + BB"

* Additive Model    : "AA" vs "AB" vs "BB"


## ML ways

### Classics Encoders

* ordinal - convert string labels to integer values 1 through k. Ordinal.

* oneHot  - one column for each value to compare vs. all other values. Nominal, ordinal.

* Binary  - convert each integer to binary digits. Each binary digit gets one column. Some info loss but fewer dimensions. Ordinal.
            (Dominant, Recessive model)

* BaseN   - Ordinal, Binary, or higher encoding. Nominal, ordinal. Doesn’t add much functionality. Probably avoid.
            (SNP-syntax)

* Hashing - Like OneHot but fewer dimensions, some info loss due to collisions. Nominal, ordinal.

* One-hot  : "all genotypes are independent"

* hashing


**Contrast Encoder**

* Helmert (reverse) : The mean of the dependent variable for a level is compared to the mean of the dependent variable over all previous levels.

* Sum : compares the mean of the dependent variable for a given level to the overall mean of the dependent variable over all the levels

* Backward Difference - the mean of the dependent variable for a level is compared with the mean of the dependent variable for the prior level. 

* Polynomial : orthogonal polynomial contrasts. The coefficients taken on by polynomial coding for k=4 levels are the linear, quadratic, and cubic trends in the categorical variable.

## association

* 
*
*

## Haplotype 

## References

* Feifei Zhao, Manshu Song, Youxin Wang, and Wei Wang (2016) **Genetic model** J Cell Mol Med.doi:10.1111/jcmm.12751
