---
title: "Genotyping"
author: "bjkim"
date: "2016년 8월 26일"
output: 
  html_document: 
    toc: true
    toc_float: true
    toc_level: 3
    highlight: tango

---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction to genotyping issues

- A good SNP from Affy 500K.

> Well clustered SNP img.
> each point - indovodual
> outside clusters means that common homozygous and rare homozygous
> center cluster means the heterozygous.

> PTPN22 in the WTCCC
> Type 1 Diabetes have more rare homozygous allels than controls.


## Overview of difficulties
- A difficulty Affy 500k SNP
- separation of intensity is not enough.
- calling genotype between lower intensities are challenging
- Effect of Copy number variation
    - Copy number variation (common deletion / )generates noisy like clusters and it inhibits normal clustering
  
## Statistical framework

# Calling strategy

# Calling strategy

## One or two dimensional clustering

## Estimation techniques

## Additional refinements

# Differental bias, informative missingness




## Illumina Platform

### Gencall

- algorithm vendor provided

- see the details in [technote_gencall_software](https://www.illumina.com/Documents/products/technotes/technote_gencall_data_analysis_software.pdf)

**Gen Train score** designed to mimic evaluation made by a human expert's visual and cognitive system.

GenCall score (GC) <0.15 is recommended threshold for failed genotype by Illumina. 

### Illuminus

- get 

### GenoSNP



## Affymetrix Platform

### 

### CRLMM ()

- extracts and summarized X and Y intensities directly from the idat files.



### References

https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-68
