---
title: "02Sequencing"
author: "bjkim"
date: '2017 5 24 '
output:
  html_document:
    toc: true
    toc_float: true
    toc_depth: 2
    number_sections: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Sequencing

## Classicial Sequencing

- Sanger Sequencing
- Maxam-Gilbert Sequencing

## Next Generation Sequencing

https://www.ebi.ac.uk/training/online/course/ebi-next-generation-sequencing-practical-course/what-you-will-learn/what-next-generation-dna-

comparing to the classical sanger sequencing, NGS are advantageous in following 4 criteria :

- speed
- cost
- sample size
- accuracy

Here are ~~~ used NGS platforms

- Illumina (Solexa) Sequencing

- Roche 454 Sequencing

- Ion torrent - proton/PGM Sequencing

- SOLiD Sequencing



### Illimina Sequencing

    * Flow cell - 8 lane . we put samples into each of the lane.
  
    * Bridge Amplification
        * Prepared Sample DNA with adaptors. the adaptors are used to attach to the solid
    
    * forming 100 - 200 molecule clusters
    
    * adding labeled nucleotides (4 Colours).

## The Sequence Read Archive (SRA)

## Genome Assembly

## Variation - The 1000 genome Project

## ChIP-seq analysis

## RAN sequeencing

## File Formats

- FASTQ/SFF
- SAM/BAM
- VCF
- FASTA
- GFF/BED

### FASTQ

The fastq format was developed to provide a convenient way of storing the sequence and the quality scores in the same file. These are text files and they look like:

```
@Read_ID_1
CTGATGTGCCGCCTCACTTCGGTGGT
+
@@@DDDDH8<BAHG@BHGIHIII>(
```

The four lines

1. the name/ID of the read, precededed by a "@". For read pair, there will be two entries with that name, either in the same or a second FASTQ file

2. the sequence of the read

3. A "+" Sign. in very old FASTQ files, this is followed by the read name from the first line. today, this line is present for historical rasons backwards compatibility only.

4. The quality scores of the bases from line 2. The scores are generated by the sequencing machine, the encoded as ASCII (33+/64) scores. the line should have the same length as line 2, as there is one quality score per base.


### SRA

SRA is the file format in which all NCBI SRA content is provided. SRA files are binary files and we need specific tools to extract the information. There is a toolkit (SRA Toolkit)developed by NCBI to deal with these binary files.
