# plink file format

This file describes various plink input/output files.

## data formats

**[Genotype files](#Genotype-files)** : 

* [pedigree and map file](#pedmap) ([.ped](#ped)/[.map](#map))
* [transposed plink file](#tpedtfam) ([.tped](#tped)/[.tfam](#tfam))
* [binary plink file](#bedbimfam) ([.bim](#bim)/[.bed](#bed)/[.fam](#fam))

**[Analysis files](#AnalysisFiles)** : 

* basic statistics 
  * frequency   : [.frq](#frq) | [.frqx](#frqx) |
  * heterozygosity : [.het](#het) | [.hh](#hh)
  * ibd/ibs     : [.genome](#genome)
  * missing     : [.imiss](#imiss) | [.lmiss](#lmiss)
* pca           : [.eigenvalue](#eigenvalue) | [.eigenvector](#eigenvector)
* association : [.assoc](#assoc)|[.assoc.adjusted](#assocadjusted) [.assoc.linear](#assoclinear)

## Genotype files

### .ped/.map

#### .ped 

It is a headerless pedigree file consisting of 6 informative columns and genotype variants; the total column is  6 + 2 * variants. A variant is written in concatenating two columns after 6th column. First 6 columns are same as the [.fam](#fam) file.

* family ID
* individual ID
* father ID
* mother ID
* sex 
  * 1 : male
  * 2 : female
  * 0/-9 : unknown/NA
* phenotype 
  * 1 : control
  * 2 : case
* genotypes (from 7th column. concatenating 2 column forms one genotype at a locus.)
  * 0    : nocall
  * A/C/G/T : allele nucleotides.

e.g) test.ped

| family  | individual | father | mother | sex | phenotype | rs1_a1 | rs1_a2 | rs...|
| :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| A | A001 |    0 |    0 | 1 | 1 | A | A | ... | 
| A | A002 |    0 |    0 | 2 | 1 | A | C | ... | 
| A | A003 | A001 | A002 | 1 | 1 | A | A | ... | 
| A | A004 | A001 | A002 | 1 | 1 | A | C | ... | 

#### .map 

map file is headerless genomic coordination file consisting of following 4 columns.
delimeter is "white space" or "tab"

* chr : chromosome number
  * 1-22: autosome
  * 23:X 
  * 24:Y 
  * 25:XY 
  * 26:MT
* marker/snp name
* morgan
* base position (bp)

e.g) test.map

| chr | marker/snp | morgan | base position |
| :---: | :---: | :---: | :---: |
| 1 | rs568121721 | 0 | 100292476 |
| 1 | rs573946207 | 0 | 101064936 |
| 1 | rs577266494 | 0 | 103380393 |

### .bed/.bim/.fam

#### .bed

Plink binary genotype table which represents genotype calls. It is different from the BED file format of the UCSC genome browser. the file starts with three magic number 0x6c, 0x1b and 0x01 in that order (plink 1.9).
You can see the `.bed` file using `xxd` UNIX command.

each two-bit gnotype codes have followng meanings

  * 00 homozygous for first allele in .bim file
  * 01 Missing genotype
  * 10 heterozygous
  * 11 homozygous for second allele in .bim file.

e.g) test.bed

**0x6c 0x1b 0x01** 0xdc 0x0f 0xe7 0x0f 0x6b 0x01

#### .bim

* bim file is headerless genomic coordination file for binary ped file (bed) consisting of following 6 columns.

* since the bed file is only written as 0 and 1, it contains allele informations at 5th and 6th columns.

Column description

* chr : chromosome number (23:X, 24:Y, 25:XY, 26:MT)

* marker/snp name

* morgan

* base position (bp)

* a1 - Allele 1. usually minor allele

* a2 - Allele 2. usually major allele


e.g ) test.bim

| chr | marker/snp | morgan | base position | a | A |
| :---: | :---: | :---: | :---: | :---: | :---: |
| 1 | rs568121721 | 0 | 100292476 | G | A |
| 1 | rs573946207 | 0 | 101064936 | G | A |
| 1 | rs577266494 | 0 | 103380393 | A | G |

#### .fam

* family ID
* individual ID
* father ID
* mother ID
* sex 
  * 1 : male
  * 2 : female
  * 0/-9 : unknown/NA
* phenotype 
  * 1 : control
  * 2 : case



#### .tped/.tfam

transposed plink file from .ped /.map

## Analysis Files

### Basic statistics

#### .frq

`.freq` is generated by plink option `--freq` and contrains allele frequency information with following 6 columns.

* CHR	Chromosome code (see [.bim](#.bed/.bim/.fam) for details)

* SNP	Variant identifier
 
* A1	Allele 1 (usually minor)

* A2	Allele 2 (usually major)

* MAF	Allele 1 frequency

* NCHROBS	Number of allele observations

e.g) test.frq

### .freqx

### .frq.count

### .het

* Short introduction
 
  Individuals within a population should have predictable heterozygosity (H) value since the individual is generated by random mating.

  If there is a large departure from expected H value, it could be sample has a quality issue or is from a different population.

* `.het` file is generated by `--het` option.

A text file with a header line, and one line per sample with the following six fields:

* FID : Family ID
* IID :	Within-family ID
* O(HOM) : Observed number of homozygotes ($`O_{HOM}`$)
* E(HOM) : Expected number of homozygotes ($`E_{HOM}`$)
* N(NM)	Number of non-missing autosomal genotypes ($`N_{NM}`$)
* F	Method-of-moments F coefficient estimate

![eq1](http://www.sciweavers.org/upload/Tex2Img_1541029677/render.png)

```math
  F = \frac{O_{HOM}-E_{HOM}}{N_{NM}-E_{HOM}}
```

e.g) 

|         FID |          IID |       O(HOM) |     E(HOM)    |    N(NM)  |         F    |
| :---------: | :----------: | :----------: | :-----------: | :-------: | :----------: |
|  AZ-17-0001 |  AZ-17-0001  |      320336  |   3.208e+05   |    417967 |  -0.005069   |
|  AZ-17-0002 |  AZ-17-0002  |      320074  |   3.213e+05   |    418536 |    -0.0122   |
|  AZ-17-0003 |  AZ-17-0003  |      321797  |   3.213e+05   |    418500 |   0.005466   |


### missing

### Association

#### .assoc.linear/.assoc.logistic

This files are 9 column multi-covariate association analysis report which generated by `--linear` or `--logistic` flags. 

```
 plink --bfile [:data:] --linear
 plink --bfile [:data:] --logistic
```


Here is the column descriptions for the output

* CHR - chromosome code
* SNP - Variant identifider
* BP - Base-pair coordinate
* A1 - usually minor allele.
* TEST - TEST identifier
* NMISS - number of observation(non-missing genotype,phenotype and covariates)
* BETA/OR - regression coefficient/odds ratio
* STAT - T-statistics
* P    - Asymptotic p-value for t-statistics

If --ci 0.xy has been specified, 3 columns are inserted before `stat`:

* SE	Standard error of beta (log-odds) estimate
* Lxy	Bottom of xy% symmetric approx. confidence interval
* Hxy	Top of xy% approx. confidence interval
